File: android/src/main/java/com/lightningkite/kwift/observables/EventToObservableProperty.actual.kt
package com.lightningkite.kwift.observables

import com.lightningkite.kwift.Box
import com.lightningkite.kwift.swiftExactly
import io.reactivex.Observable


fun <T> Observable<Box<T>>.asObservablePropertyUnboxed(defaultValue: T): ObservableProperty<T> {
    return EventToObservableProperty<T>(defaultValue, this)
}

File: ios/Khrysalis/com/lightningkite/khrysalis/observables/EventToObservableProperty.actual.swift
import RxSwift
import RxRelay

//--- Observable<T>.asObservablePropertyUnboxed(T)
extension Observable {
    public func asObservablePropertyNullable(defaultValue: Element) -> ObservableProperty<Element> {
        return EventToObservableProperty<Element>(defaultValue, self)
    }
    public func asObservablePropertyNullable(_ defaultValue: Element) -> ObservableProperty<Element> {
        return asObservablePropertyNullable(defaultValue: defaultValue)
    }
}
